src/snowflake/cli/_plugins/cortex/commands.py

```diff
@@ -152,14 +169,13 @@ def extract_answer(
         help="String containing the question to be answered.",
         show_default=False,
     ),
-    source_document_text: Optional[str] = typer.Argument(
-        None,
+    source_document_text: Optional[str] = TextSourceArgument(
```

Comment:
having the name of the argument directly associated with SOURCE_EXCLUSIVE_OPTION_NAMES seems like an antipattern. 
If using (Exclusive)TextSourceArgument and (Exclusive)ReadableFileOption is not enough to ensure exclusivity, then I believe this change is not very useful, as you still need to explicitly specify what fields are exclusive.
Example issues:
- new command that uses new names.
- a command that does not require exclusivity
- command that was using TextSourceArgument (not for the purpose of exclusivity), but suddenly becomes exclusive on rename.

Reply:
Thank you!  I rename both arguments to ExclusiveTextSourceArgument and ExclusiveReadableFileOption.

In some cases, the exclusive relationship between parameters is not transitive. Defining exclusiveness solely based on the parameter type can be less useful, as it assumes transitivity at all times.